src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Methodologies/MethodologyService.cs

```diff
@@ -200,6 +256,50 @@ private async Task DeleteMethodologyParentIfOrphaned(Methodology methodology)
             }
         }
 
+        private async Task<Either<ActionResult, Unit>> CheckMethodologyCanDependOnRelease(
+            Methodology methodology,
+            MethodologyUpdateRequest request)
+        {
+            if (request.PublishingStrategy != WithRelease)
+            {
+                return Unit.Instance;
+            }
+
+            if (!request.WithReleaseId.HasValue)
+            {
+                return new NotFoundResult();
+            }
+
+            // Check that this release exists, that it's not already published, and that it's using the methodology
+            return await _persistenceHelper.CheckEntityExists<Release>(request.WithReleaseId.Value)
+                .OnSuccess(async release =>
+                {
+                    if (release.Live)
+                    {
+                        throw new ArgumentException("Methodology cannot depend on a published Release");
```

Comment:
Yeah, so here Id be tempted to return a BadResult, unless the thowing of the Exception does some DbContext-unwinding or something?

Reply:
I've changed this and also the one below now, thanks.